ext/standard/pack.c

```diff
@@ -1043,8 +1044,14 @@ PHP_FUNCTION(unpack)
 
 					case 'i':   /* signed integer, machine size, machine endian */
 					case 'I': { /* unsigned integer, machine size, machine endian */
-						unsigned int x = *((unaligned_uint*) &input[inputpos]);
-						zend_long v = (type == 'i') ? (int) x : x;
+						zend_long v;
+						if (type == 'i') {
+							int x = *((unaligned_int*) &input[inputpos]);
+							v = x;
+						} else {
+							unsigned int x = *((unaligned_uint*) &input[inputpos]);
+							v = x;
```

Comment:
`v = x;` isn't better after the else {}?

Reply:
x doesn't exist anymore after the else.